[docs] add logical delete documentation (#163)
OpenCTI-Platform/docs: docs/usage/delete-restore.md

Reviewer: Could we also explain the impact on te streams of the restore? Something like: "When restoring, it acts like a create in the platform. Therefore, your feeds will be impact following the usual creation process. 
A restored item will have a specific record within its activity, called "restore"".

Reviewer: Not sure where to put it in the file, hence only a comment.

# Delete and restore knowledge

Knowledge can be deleted from OpenCTI either in an overview of an object or using [background tasks](background-tasks.md)

> When we delete an object, we also delete all its relationships and references to other objects. 

Reviewer 2: ```suggestion
When an object is deleted, all its relationships and references to other objects are also deleted. 
```

The deletion event is written to the [stream](../reference/streaming.md), to trigger automated [playbooks](./automation.md) or synchronize another platform.

Prior to 6.1 version, there was no way to undo this operation and the knowledge was lost permanently.

> Since OpenCTI 6.1, we now keep record of the deleted objects for a given period of a time, allowing to restore them on demand. This does not impact the stream events or other side effect of the deletion: the object is still _deleted_.

Reviewer 2: ```suggestion
Since OpenCTI 6.1, a record of all the deleted objects is kept for a given period of time, allowing to restore them on demand. This does not impact the stream events or other side effect of the deletion: the object is still _deleted_.
```

## Trash

> We added a view called "Trash" that will display all delete operations, entities and relationships alike.

Reviewer 2: ```suggestion
A view called "Trash" displays all "delete" operations, entities and relationships alike.
```

![Trash](assets/trash.png)

A delete operation contains not only the entity or relationship that has been deleted, but also all the relationships and references from (to) this main object to (from) other elements in the platform.

You can sort, filter or search this table using the usual UI controls. You are limited to the type of object, their representation (most of the time, the _name_ of the object), the user who deleted the object, the date and time of deletion and the marking of the object.

Note that the delete operations (i.e. the entries in this table view) inherit the marking of the main entity that was deleted, and thus follow the same access restriction as the object that was deleted.

You can individually restore or permanently delete an object from the trash view using the burger menu at the end of the line.

![Trash actions](assets/trash-actions.png)

Alternatively, you can use the checkboxes at the start of the line to select a subset of deleted objects, and trigger a background task to restore or permanently delete them by batch.

## Restore

Restoring an element will create it again in the platform with the same information it had before it has been deleted.

> It will also restore all the relationships from or to this object, that have been also deleted when deleting the object.

Reviewer 2: ```suggestion
It also restores all the relationships from or to this object, that have been also deleted during the deletion operation.
```

If the object had attached files (uploaded or exported), they will be also restored.

![Trash restore confirm](assets/trash-restore-confirm.png)

## Permanent delete

> We can also delete permanently the object, its relationships, and attached files.

Reviewer 2: ```suggestion
From the Trash panel, it is also possible to delete permanently the object, its relationships, and attached files.
```
